indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -156,7 +158,11 @@ public class ParallelIndexSupervisorTask extends AbstractBatchIndexTask implemen
    * Only the compaction task can have a special base name.
    */
   private final String baseSubtaskSpecName;
+
   private final InputSource baseInputSource;
+  // A reference to the baseInputSource or another input source craeted using the baseInputSource and a TaskToolbox
+  // This helps return the same object
+  private final AtomicReference<InputSource> inputSource = new AtomicReference<>();
```

Comment:
Maybe call it `inputSourceWithToolbox` for clarity.

Reply:
Done